pippy/IR.py

```diff
@@ -1005,6 +1005,21 @@ def move_param_to_callee(
 
         split.delete_all_unused_submodules()
 
+        # Users want the first pipeline stage to accept kwargs if the original
+        # program does. This is controlled by the `_codegen` field of the graph,
+        # so we make a copy here. Note: we only want the input spec and not the
+        # output spec, because the output spec is for the last stage. Maybe a
+        # TODO? Not sure yet.
+        submod0 = list(split.children())[0]
+        submod0.graph._codegen = copy.deepcopy(traced.graph._codegen)
+        # `_replace` is actually not "private" or internal. based on this doc:
+        # To prevent conflicts with field names, the method and attribute names
+        # start with an underscore
+        submod0.graph._codegen.pytree_info = (
+            submod0.graph._codegen.pytree_info._replace(out_spec=None)
```

Comment:
why do you want to set it to be None? Then the output structure will not match the eager module -- but not sure if you care about this.

Reply:
Because the original program's output format should be applied to the last stage, not stage 0 :)